### PR TITLE
[Improve] use generate_sink_sql=true for db2 upsert

### PR DESCRIPTION
**Pull request by shashwatsai**
## Purpose of this pull request
Currently we enforce sending sink insert query, forcing user to not support upsert scenarios. 
Please refer: 
ISSUE: https://github.com/apache/seatunnel/issues/4434
PR (Upsert DB2 Support): https://github.com/apache/seatunnel/pull/7879

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-server/seatunnel-app/src/main/java/org/apache/seatunnel/app/thirdparty/datasource/impl/Db2DataSourceConfigSwitcher.java [hunk 1]
```diff
@@ -110,13 +110,15 @@ public Config mergeDatasourceConfig(
...
                         connectorConfig.withValue(QUERY_KEY, ConfigValueFactory.fromAnyRef(sql));
             } else if (pluginType.equals(PluginType.SINK)) {
-
-                List<String> tableFields = selectTableFields.getTableFields();
-
-                String sql = generateDb2(tableFields, databaseName, tableName);
-
-                connectorConfig =
-                        connectorConfig.withValue(QUERY_KEY, ConfigValueFactory.fromAnyRef(sql));
+                if (selectTableFields.isAll()) {
```
Comment (arshadmohammad) [inline seatunnel-server/seatunnel-app/src/main/java/org/apache/seatunnel/app/thirdparty/datasource/impl/Db2DataSourceConfigSwitcher.java:113]: I do not see any reference of generate_sink_sql change or use. an you explain the code change, how it is related to db2 upsert

File: seatunnel-server/seatunnel-app/src/main/java/org/apache/seatunnel/app/thirdparty/datasource/impl/Db2DataSourceConfigSwitcher.java [hunk 2]
```diff
@@ -110,13 +110,15 @@ public Config mergeDatasourceConfig(
...
+                    // Convert schema to SQL if all fields are present
+                    // else use default generate_sink_sql=true
+                    List<String> tableFields = selectTableFields.getTableFields();
+                    String sql = generateDb2(tableFields, databaseName, tableName);
+                    connectorConfig =
+                            connectorConfig.withValue(
+                                    QUERY_KEY, ConfigValueFactory.fromAnyRef(sql));
+                }
```
Comment (arshadmohammad) [inline seatunnel-server/seatunnel-app/src/main/java/org/apache/seatunnel/app/thirdparty/datasource/impl/Db2DataSourceConfigSwitcher.java:121]: can you attach snapshot or test cases for this change

**Comment by shashwatsai** [comment]
Manual Deployment:

Source: Hive JDBC (partitioned table)
image

Sink: DB2
image

Trigger for row updates:
image

generated job config:
image

Validations: 
Difference for keys: 745, 766, 779
image

**Comment by arshadmohammad** [comment]
Thanks @shashwatsai  for working on db2 issues.

**Pull request merged**